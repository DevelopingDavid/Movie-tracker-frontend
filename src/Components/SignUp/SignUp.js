import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';

class SignUp extends Component {
  constructor() {
    super();
    this.state = {
      email: '',
      password: '',
      name: '',
      error: ''
    }
  }

  addUserToDatabase = async (event) => {
    event.preventDefault();
    const userInfo = this.state;
    const options = {
      method: "post",
      headers: {
        "Content-Type": "application/json"
      },
      body: JSON.stringify(userInfo)
    }
    const userDatabase = 'http://localhost:3000/api/users/new';
    const response = await fetch(userDatabase, options);
    const data = await response.json();
  }

  saveInput = (event) => {
    event.preventDefault();
    const {name, value} = event.target;
    this.setState({
      [name]: value
    });
  }

  render() {
    return(
      <div>
        <form className='sign-up' onChange={this.saveInput} onSubmit={this.addUserToDatabase}>
          <div>
            <label>Email</label>
            <i className="fas fa-user"></i>
          </div>
          <input name='name' type='text' placeholder='Name' autoComplete="off"/>
          <div>
            <label>Email</label>
            <i className="fas fa-envelope"></i>            
          </div>
          <input name='email' type='email' placeholder='email@example.com' autoComplete="off"/>
          <div>
            <label>Password</label>
            <i className="fas fa-lock"></i>          
          </div>
          <input name='password' type='password' placeholder='password' autoComplete="off"/>
          <input type='submit'/>
        </form>
        <NavLink to='/sign-in'>Already have an account? Sign in</NavLink>        
      </div>
    )
  }
}

export default SignUp