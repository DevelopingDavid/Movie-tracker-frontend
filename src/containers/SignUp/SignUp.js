import React, { Component } from 'react';
import { NavLink, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import { loginUser } from '../../actions';
import PropTypes from 'prop-types';

export class SignUp extends Component {
  constructor() {
    super();
    this.state = {
      email: '',
      password: '',
      name: '',
      error: ''
    }
  }

  addUserToDatabase = async (event) => {
    event.preventDefault();
    const userInfo = this.state;
    const options = {
      method: "post",
      headers: {
        "Content-Type": "application/json"
      },
      body: JSON.stringify(userInfo)
    }
    const userDatabase = 'http://localhost:3000/api/users/new';
    const response = await fetch(userDatabase, options);
    if(response.ok) {
      const data = await response.json();
      this.props.loginUser(data.id, this.state.name)
      return data;
    } else {
      return 'Please fill out form'
    }
  }

  saveInput = (event) => {
    event.preventDefault();
    const {name, value} = event.target;
    this.setState({
      [name]: value
    });
  }

  render() {

    if(Object.keys(this.props.user).length > 0) { 
      return <Redirect to='/movies'/>
    }

    return(
      <div className='sign-up-container'>
        <section>
          <form className='sign-up-form' onChange={this.saveInput} onSubmit={this.addUserToDatabase}>
            <div>
              <label>Name</label>
              <i className="fas fa-user"></i>
            </div>
            <input name='name' type='text' placeholder='Name' autoComplete="off"/>
            <div>
              <label>Email</label>
              <i className="fas fa-envelope"></i>            
            </div>
            <input name='email' type='email' placeholder='email@example.com' autoComplete="off"/>
            <div>
              <label>Password</label>
              <i className="fas fa-lock"></i>          
            </div>
            <input name='password' type='password' placeholder='password' autoComplete="off"/>
            <input className="sign-up-btn" type='submit'/>
          </form>
          <NavLink className='sign-in-link' to='/sign-in'>Already have an account? Sign in</NavLink>        
        </section>
      </div>
    )
  }
}

export const mapDispatchToProps = (dispatch) => ({
  loginUser: (id, name) => dispatch(loginUser(id, name))
});

export const mapStateToProps = (state) => ({
  user: state.user
});

export default connect(mapStateToProps, mapDispatchToProps)(SignUp);

SignUp.propTypes = {
  loginUser: PropTypes.func.isRequired,
  user: PropTypes.object.isRequired
}