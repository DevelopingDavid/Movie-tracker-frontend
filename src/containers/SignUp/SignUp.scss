.sign-up-container {
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
  section {
    height: 80%;
    background-color: #0e0f11;
    width: 50%;
    color: white;
    border-radius: 1.7rem;
    font-family: sans-serif;
    font-size: 2rem;
    border: solid 3px #940002;
  }
  form {
    margin: 100px 100px 0 100px;
  }
  input {
    margin: 20px 0;
    border: none;
    border-bottom: solid 2px white;
    background: none;
    font-size: 2rem;
    width: 100%;
    outline: none;
    padding-bottom: 5px;
    color: white;
  }
  .sign-up-btn {
    background-color: white;
    color: black;
    & :hover {
      color: pink;
    }
  }
  label {
    margin-right: 14px;
  }
}

.sign-up-form {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
}

.sign-in-link {
  margin-top: 50px;
  color: white;
}
