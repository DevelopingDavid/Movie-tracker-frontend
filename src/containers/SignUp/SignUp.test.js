import React from 'react';
import SignUp from '../SignUp/SignUp';
import { shallow } from 'enzyme';

describe('SignUp', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow( 
      <SignUp/>
    ) 
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should start with default state', () => {
    expect(wrapper.state()).toEqual({
      email: '',
      password: '',
      name: '',
      error: ''
    })
  });
});