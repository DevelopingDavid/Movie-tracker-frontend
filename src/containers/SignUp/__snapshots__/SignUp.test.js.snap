// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUp />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <section>
        <form
          className="sign-up-form"
          onChange={[Function]}
          onSubmit={[Function]}
        >
          <div>
            <label>
              Name
            </label>
            <i
              className="fas fa-user"
            />
          </div>
          <input
            autoComplete="off"
            name="name"
            placeholder="Name"
            type="text"
          />
          <div>
            <label>
              Email
            </label>
            <i
              className="fas fa-envelope"
            />
          </div>
          <input
            autoComplete="off"
            name="email"
            placeholder="email@example.com"
            type="email"
          />
          <div>
            <label>
              Password
            </label>
            <i
              className="fas fa-lock"
            />
          </div>
          <input
            autoComplete="off"
            name="password"
            placeholder="password"
            type="password"
          />
          <input
            className="sign-up-btn"
            type="submit"
          />
        </form>
        <NavLink
          className="sign-in-link"
          to="/sign-in"
        >
          Already have an account? Sign in
        </NavLink>
      </section>,
      "className": "sign-up-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            className="sign-up-form"
            onChange={[Function]}
            onSubmit={[Function]}
          >
            <div>
              <label>
                Name
              </label>
              <i
                className="fas fa-user"
              />
            </div>
            <input
              autoComplete="off"
              name="name"
              placeholder="Name"
              type="text"
            />
            <div>
              <label>
                Email
              </label>
              <i
                className="fas fa-envelope"
              />
            </div>
            <input
              autoComplete="off"
              name="email"
              placeholder="email@example.com"
              type="email"
            />
            <div>
              <label>
                Password
              </label>
              <i
                className="fas fa-lock"
              />
            </div>
            <input
              autoComplete="off"
              name="password"
              placeholder="password"
              type="password"
            />
            <input
              className="sign-up-btn"
              type="submit"
            />
          </form>,
          <NavLink
            className="sign-in-link"
            to="/sign-in"
          >
            Already have an account? Sign in
          </NavLink>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <label>
                  Name
                </label>
                <i
                  className="fas fa-user"
                />
              </div>,
              <input
                autoComplete="off"
                name="name"
                placeholder="Name"
                type="text"
              />,
              <div>
                <label>
                  Email
                </label>
                <i
                  className="fas fa-envelope"
                />
              </div>,
              <input
                autoComplete="off"
                name="email"
                placeholder="email@example.com"
                type="email"
              />,
              <div>
                <label>
                  Password
                </label>
                <i
                  className="fas fa-lock"
                />
              </div>,
              <input
                autoComplete="off"
                name="password"
                placeholder="password"
                type="password"
              />,
              <input
                className="sign-up-btn"
                type="submit"
              />,
            ],
            "className": "sign-up-form",
            "onChange": [Function],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <label>
                    Name
                  </label>,
                  <i
                    className="fas fa-user"
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Name",
                  },
                  "ref": null,
                  "rendered": "Name",
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-user",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "autoComplete": "off",
                "name": "name",
                "placeholder": "Name",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <label>
                    Email
                  </label>,
                  <i
                    className="fas fa-envelope"
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Email",
                  },
                  "ref": null,
                  "rendered": "Email",
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-envelope",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "autoComplete": "off",
                "name": "email",
                "placeholder": "email@example.com",
                "type": "email",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <label>
                    Password
                  </label>,
                  <i
                    className="fas fa-lock"
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Password",
                  },
                  "ref": null,
                  "rendered": "Password",
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-lock",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "autoComplete": "off",
                "name": "password",
                "placeholder": "password",
                "type": "password",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "sign-up-btn",
                "type": "submit",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Already have an account? Sign in",
            "className": "sign-in-link",
            "to": "/sign-in",
          },
          "ref": null,
          "rendered": "Already have an account? Sign in",
          "type": [Function],
        },
      ],
      "type": "section",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <section>
          <form
            className="sign-up-form"
            onChange={[Function]}
            onSubmit={[Function]}
          >
            <div>
              <label>
                Name
              </label>
              <i
                className="fas fa-user"
              />
            </div>
            <input
              autoComplete="off"
              name="name"
              placeholder="Name"
              type="text"
            />
            <div>
              <label>
                Email
              </label>
              <i
                className="fas fa-envelope"
              />
            </div>
            <input
              autoComplete="off"
              name="email"
              placeholder="email@example.com"
              type="email"
            />
            <div>
              <label>
                Password
              </label>
              <i
                className="fas fa-lock"
              />
            </div>
            <input
              autoComplete="off"
              name="password"
              placeholder="password"
              type="password"
            />
            <input
              className="sign-up-btn"
              type="submit"
            />
          </form>
          <NavLink
            className="sign-in-link"
            to="/sign-in"
          >
            Already have an account? Sign in
          </NavLink>
        </section>,
        "className": "sign-up-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <form
              className="sign-up-form"
              onChange={[Function]}
              onSubmit={[Function]}
            >
              <div>
                <label>
                  Name
                </label>
                <i
                  className="fas fa-user"
                />
              </div>
              <input
                autoComplete="off"
                name="name"
                placeholder="Name"
                type="text"
              />
              <div>
                <label>
                  Email
                </label>
                <i
                  className="fas fa-envelope"
                />
              </div>
              <input
                autoComplete="off"
                name="email"
                placeholder="email@example.com"
                type="email"
              />
              <div>
                <label>
                  Password
                </label>
                <i
                  className="fas fa-lock"
                />
              </div>
              <input
                autoComplete="off"
                name="password"
                placeholder="password"
                type="password"
              />
              <input
                className="sign-up-btn"
                type="submit"
              />
            </form>,
            <NavLink
              className="sign-in-link"
              to="/sign-in"
            >
              Already have an account? Sign in
            </NavLink>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div>
                  <label>
                    Name
                  </label>
                  <i
                    className="fas fa-user"
                  />
                </div>,
                <input
                  autoComplete="off"
                  name="name"
                  placeholder="Name"
                  type="text"
                />,
                <div>
                  <label>
                    Email
                  </label>
                  <i
                    className="fas fa-envelope"
                  />
                </div>,
                <input
                  autoComplete="off"
                  name="email"
                  placeholder="email@example.com"
                  type="email"
                />,
                <div>
                  <label>
                    Password
                  </label>
                  <i
                    className="fas fa-lock"
                  />
                </div>,
                <input
                  autoComplete="off"
                  name="password"
                  placeholder="password"
                  type="password"
                />,
                <input
                  className="sign-up-btn"
                  type="submit"
                />,
              ],
              "className": "sign-up-form",
              "onChange": [Function],
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <label>
                      Name
                    </label>,
                    <i
                      className="fas fa-user"
                    />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Name",
                    },
                    "ref": null,
                    "rendered": "Name",
                    "type": "label",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "fas fa-user",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "autoComplete": "off",
                  "name": "name",
                  "placeholder": "Name",
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <label>
                      Email
                    </label>,
                    <i
                      className="fas fa-envelope"
                    />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Email",
                    },
                    "ref": null,
                    "rendered": "Email",
                    "type": "label",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "fas fa-envelope",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "autoComplete": "off",
                  "name": "email",
                  "placeholder": "email@example.com",
                  "type": "email",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <label>
                      Password
                    </label>,
                    <i
                      className="fas fa-lock"
                    />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Password",
                    },
                    "ref": null,
                    "rendered": "Password",
                    "type": "label",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "fas fa-lock",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "autoComplete": "off",
                  "name": "password",
                  "placeholder": "password",
                  "type": "password",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "sign-up-btn",
                  "type": "submit",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "form",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Already have an account? Sign in",
              "className": "sign-in-link",
              "to": "/sign-in",
            },
            "ref": null,
            "rendered": "Already have an account? Sign in",
            "type": [Function],
          },
        ],
        "type": "section",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
